datapackage/resource.py

```diff
@@ -279,6 +279,7 @@ def iter(self):
                 parser_class = tabulator.parsers.CSV
             try:
                 table = tabulator.topen(data_path_or_url, with_headers=True,
+                                        encoding=self.metadata.get('encoding'),
```

Comment:
Yeah, this is one of the places where we have to decide if we're strict with the spec or not. I don't like leaving that decision to `tabulator`, but that's the best we can do IMO.

I like your test idea. I'm not sure which exception should be thrown in that case (I'd guess `ValueError`). Just make sure you update the docstrings.